### ISSUE TITLE
Confusion about the implementation of accelerate method in SlidingAgent

### ISSUE THREAD

**Issue by Hcnaeg** [thomashirtz/gym-hybrid#6]
Hello, I am kind of confused about the implementation of the accelerate method [at here](https://github.com/thomashirtz/gym-hybrid/blob/2c2eb3962d346ef6be2aed209020e47c1da285d4/gym_hybrid/agents.py#L59)
According to the following formula from [stackexchange](https://math.stackexchange.com/a/1365938/849658) (as mentioned in L58),
<img width="271" alt="image" src="https://user-images.githubusercontent.com/51738591/168269408-82f648b1-fa5c-4cbf-934f-c056061a39f9.png">
to calculate the magnitude of the sum of to polar vectors, we need to know the magnitudes and angles of these two polar vectors, and I think `self.speed` and `speed` represent the magnitudes, but I don't understand why they're used to calculate the cosine value by `np.cos(value - self.speed)` (instead of something represents the angle). By the way, I don't quite understand the way you simulate inertia, from my physics knowledge, it may be necessary to use some variables to represent the 'acceleration' or 'derivative' which specifies how fast the speed changes, am I right ?

Also, I wonder what does "direction of the agent" mean at [here](https://github.com/thomashirtz/gym-hybrid/blob/2c2eb3962d346ef6be2aed209020e47c1da285d4/gym_hybrid/agents.py#L9) and what's the difference between "direction of the agent" and "angle of the velocity vector".

Your timely response will be greatly appreciated.

**Comment by thomashirtz** [comment]
You are right, there seems to be an issue with the magnitude, and apparently the angle too, `self.theta + self.phi*np.cos(self.phi - self.theta)` seems wrong as well. 

I started to do research on hybrid agent, I changed topic before even finishing this piece of code, I still finished it and released it as I thought it would be useful for some people, however this environment was not fully tested.

As I remember, the inertia was managed by basically increase the speed by a certain amount, during one timestep. https://github.com/thomashirtz/gym-hybrid/blob/2c2eb3962d346ef6be2aed209020e47c1da285d4/gym_hybrid/agents.py#L40

I assumed that the agent has a speed towards phi and a speed of "acceleration" (that lasts only one timestep) towards theta  (because "acceleration = delta(speed) / delta(time)" and delta t" in our case is one timestep). I will correct the two equation, however if you find some better way to handle that I would be please to hear it and you are also welcome to do a MR.

Theta is the angle of the agent with the board as reference. It means that the agent is pointing towards theta. However the velocity vector is pointing towards phi. (this means that if we accelerate, we accelerate in the direction of the agent (theta) and not the direction of the existing velocity (phi))

**Comment by Hcnaeg** [comment]
Thanks a lot for your explanation, I may think of a way to fix this issue and contact you later.
